packages/orchestrator-ui-components/src/hooks/DataFetchHooks.ts

```diff
@@ -1,90 +1,67 @@
-import { useQuery } from 'react-query';
 import { useContext } from 'react';
 import { OrchestratorConfigContext } from '../contexts/OrchestratorConfigContext';
-import { ItemsList } from '../types';
-
-async function getFavouriteSubscriptions(apiUrl: string) {
-    const response = await fetch(apiUrl + '/subscriptions/?range=10%2C15');
-    return await response.json();
-}
-async function getProcessesNeedingAttention(apiUrl: string) {
-    const response = await fetch(apiUrl + '/processes/?range=100%2C105');
-    return await response.json();
-}
-async function getRecentProcesses(apiUrl: string) {
-    const response = await fetch(apiUrl + '/processes/?range=106%2C111');
-    return await response.json();
-}
+import { ItemsList, ProcessFromRestApi } from '../types';
+import { useQueryWithFetch } from './useQueryWithFetch';
 
 export type CacheNames = { [key: string]: string };
 
-async function getCacheNames(apiUrl: string) {
-    const response = await fetch(apiUrl + '/settings/cache-names');
-    return (await response.json()) as CacheNames;
-}
-
 export const useFavouriteSubscriptions = () => {
     const { orchestratorApiBaseUrl } = useContext(OrchestratorConfigContext);
-    const { data, isLoading } = useQuery(['favouriteSubscriptions'], () =>
-        getFavouriteSubscriptions(orchestratorApiBaseUrl),
-    );
+    const url = `${orchestratorApiBaseUrl}/subscriptions/?range=10%2C15`;
     const initialData: ItemsList = {
         buttonName: 'Show all favourites',
         items: [],
         title: 'Favourite Subscriptions',
         type: 'subscription',
     };
 
-    return isLoading
-        ? initialData
-        : {
-              ...initialData,
-              items: data,
-          };
+    const { data, isLoading } = useQueryWithFetch<
+        ProcessFromRestApi[],
+        Record<string, never>
+    >(url, {}, 'favouriteSubscriptions');
+    return isLoading ? initialData : { ...initialData, items: data || [] };
 };
 
 export const useProcessesAttention = () => {
     const { orchestratorApiBaseUrl } = useContext(OrchestratorConfigContext);
-    const { data, isLoading } = useQuery(['processesAttention'], () =>
-        getProcessesNeedingAttention(orchestratorApiBaseUrl),
-    );
+    const url = `${orchestratorApiBaseUrl}/processes/?range=100%2C105`;
```

Comment:
This value seems magical though :100%2C105 Are you sure that's correct and if so can you put it into a meaningful variable?

Reply:
not my idea, only updated the way how we fetch from rest endpoints, the url existed beforehand and didn't really find it necessary to update.